static/app/views/replays/details.tsx

```diff
@@ -31,95 +26,68 @@ function ReplayDetails() {
     return (
-      <DetailLayout orgId={orgId}>
+      <Page orgId={orgId}>
         <PageContent>
           <NotFound />
         </PageContent>
-      </DetailLayout>
+      </Page>
     );
   }
 
   if (!fetching && replay && replay.getRRWebEvents().length < 2) {
     return (
-      <DetailLayout orgId={orgId}>
+      <Page orgId={orgId} event={replay.getEvent()}>
         <DetailedError
           onRetry={onRetry}
           hideSupportLinks
           heading={t('Expected two or more replay events')}
           message={
-            <React.Fragment>
+            <Fragment>
               <p>{t('This Replay may not have captured any user actions.')}</p>
               <p>
                 {t(
                   'Or there may be an issue loading the actions from the server, click to try loading the Replay again.'
                 )}
               </p>
-            </React.Fragment>
+            </Fragment>
           }
         />
-      </DetailLayout>
+      </Page>
     );
   }
 
   return (
     <ReplayContextProvider replay={replay} initialTimeOffset={initialTimeOffset}>
-      <DetailLayout orgId={orgId}>
-        <Layout.Body>
-          <Layout.Main ref={fullscreenRef}>
-            <ReplayView toggleFullscreen={toggleFullscreen} isFullscreen={isFullscreen} />
-          </Layout.Main>
-
-          <Layout.Side>
-            <ErrorBoundary mini>
-              <AsideTabs replay={replay} />
-            </ErrorBoundary>
-          </Layout.Side>
-
-          <StickyMain fullWidth>
-            <ErrorBoundary mini>
-              <ReplayTimeline />
-            </ErrorBoundary>
-            <FocusTabs />
-          </StickyMain>
-
-          <StyledLayoutMain fullWidth>
-            <ErrorBoundary mini>
-              <FocusArea />
-            </ErrorBoundary>
-          </StyledLayoutMain>
-        </Layout.Body>
-      </DetailLayout>
+      <LoadedDetails orgId={orgId} />
     </ReplayContextProvider>
   );
 }
 
-const StickyMain = styled(Layout.Main)`
-  position: sticky;
-  top: 0;
-  z-index: ${p => p.theme.zIndex.header};
-
-  /* Make this component full-bleed, so the background covers everything underneath it */
-  margin: -${space(1.5)} -${space(4)} -${space(3)};
-  padding: ${space(1.5)} ${space(4)} 0;
-  max-width: none;
-  background: ${p => p.theme.background};
-`;
-
-const StyledLayoutMain = styled(Layout.Main)`
-  ${HeaderContainer} {
-    position: relative;
-  }
-`;
+function LoadedDetails({orgId}: {orgId: string}) {
+  const {getParamValue} = useUrlParam('l_page', 'topbar');
+  const {duration, replay} = useReplayContext();
+  return (
+    <Page
+      orgId={orgId}
+      crumbs={replay?.getRawCrumbs()}
+      duration={duration}
+      event={replay?.getEvent()}
+    >
+      <Layout
+        layout={
+          // TODO(replay): If we end up keeping this, we'll fix up the typing
+          LAYOUT_NAMES.includes(getParamValue()) ? (getParamValue() as any) : 'topbar'
+        }
+      />
+    </Page>
+  );
+}
```

Comment:
-> Had to make this a component so that i could use the hooks.
- -> Need the hooks so I can pull out the `crumbs` and `duration` which are only calculated inside the context provider. 

So we have layered code like:
```
<ReplayContextProvider replay={replay}>
      <LoadedDetails /> // <- reads `replay` (and more) back from the context
</ReplayContextProvider>
```

I'd prefer if the `<Page>` was always outside of the context, then it would never break. The way it is now someone could add a 'useReplayContext()` call inside <Page> and break the loading or fetch-failed views.